packages/core/src/MenuItem/MenuItem.js

```diff
@@ -1,62 +1,97 @@
  * @returns {React.Component}
  *
- * @example import { MenuItem } from '@dhis2/ui-core'
+ * @example import { MenuItem } from '@dhis2/ui
  *
  * @see Specification: {@link https://github.com/dhis2/design-system/blob/master/molecules/menu.md|Design system}
- * @see Live demo: {@link /demo/?path=/story/menu--default|Storybook}
+ * @see Live demo: {@link /demo/?path=/story/components-core-menulist--default|Storybook}
  */
 const MenuItem = ({
     href,
-    value,
-    label,
-    icon,
+    onClick,
     children,
-    active,
+    target,
+    icon,
+    className,
     destructive,
     disabled,
     dense,
-    onClick,
-    className,
-    target,
+    active,
     dataTest,
+    chevron,
+    value,
+    label,
+    showSubMenu,
+    toggleSubMenu,
 }) => {
-    return !children ? (
-        <MenuItemContent
-            active={active}
-            className={className}
-            dataTest={dataTest}
-            dense={dense}
-            destructive={destructive}
-            disabled={disabled}
-            href={href}
-            icon={icon}
-            label={label}
-            target={target}
-            value={value}
-            onClick={onClick}
-        />
-    ) : (
-        <SubMenu
-            active={active}
-            className={className}
-            dataTest={dataTest}
-            dense={dense}
-            destructive={destructive}
-            disabled={disabled}
-            icon={icon}
-            label={label}
-        >
-            {children}
-        </SubMenu>
+    const menuItemRef = useRef()
+    const { node } = useLayerContext()
+
+    return (
+        <>
+            <li
+                className={cx(className, {
+                    destructive,
+                    disabled,
+                    dense,
+                    active,
+                    'with-chevron': chevron,
+                })}
+                ref={menuItemRef}
+                data-test={dataTest}
+            >
+                <a
+                    target={target}
+                    href={!disabled && href ? href : undefined}
+                    onClick={
+                        !disabled &&
+                        createOnClickHandler(onClick, toggleSubMenu, value)
+                    }
+                >
+                    {icon && <span className="icon">{icon}</span>}
+
+                    <span className="label">{label}</span>
+
+                    {(chevron || children) && (
+                        <span className="chevron">
+                            <ChevronRight />
+                        </span>
+                    )}
+                </a>
+
+                <style jsx>{styles}</style>
+            </li>
+            {showSubMenu &&
+                createPortal(
+                    <Popper placement="right-start" reference={menuItemRef}>
+                        <Menu dense={dense}>{children}</Menu>
+                    </Popper>,
+                    node
+                )}
```

Comment:
I think we could move this into the `li`. Technically it wouldn't make any difference, but then in the code, the `li` would be the outer component instead of a fragment (easier to parse)

Reply:
Personally, I think using a fragment here signals very clearly that these components/elements don't have a parent-child relationship.

This is also in line with what we do pretty much everywhere we use a `createPortal` and `<Layer />` (which uses createPortal internally).